packages/flutter/lib/src/material/data_table.dart

```diff
@@ -830,11 +836,13 @@ class DataTable extends StatelessWidget {
     required bool ascending,
     required MaterialStateProperty<Color?>? overlayColor,
     required MouseCursor? mouseCursor,
+    required bool centerLabel,
   }) {
     final ThemeData themeData = Theme.of(context);
     final DataTableThemeData dataTableTheme = DataTableTheme.of(context);
     label = Row(
       textDirection: numeric ? TextDirection.rtl : null,
+      mainAxisSize: centerLabel ? MainAxisSize.min : MainAxisSize.max,
```

Comment:
What if we wrap _SortArrow in a Flexible below? Would that allow someone to center the label themselves without messing up the sort arrow without needing to add a `centerLabel` parameter?

Reply:
@wer-mathurin 
Thanks, i will account for the `_sortArrowPadding`. Currently, waiting for the feedback from @Piinks on this [idea](https://github.com/flutter/flutter/pull/144006#discussion_r1528467719)

Reply:
Love the mainAxisAlignment idea. More consistency is always better! :)

Reply:
I was originally thinking there could be multiple Flexibles, for the empty space on either side, the arrow and the label, but I think you have it.